source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -101,20 +103,55 @@ TextureBaker::TextureBaker(unsigned int width, unsigned int height, Image::BaseT
     _frameCaptureImage->createResourceBuffer();
 }
 
-FilePath TextureBaker::generateTextureFilename(OutputPtr output, const string& shaderName, const string& udim)
+size_t TextureBaker::findVarInTemplate(const string& filename, const string& var, size_t start)
 {
-    string outputName = createValidName(output->getNamePath());
-    string shaderSuffix = shaderName.empty() ? EMPTY_STRING : "_" + shaderName;
-    string udimSuffix = udim.empty() ? EMPTY_STRING : "_" + udim;
-    std::string bakedImageName; 
-    bakedImageName = outputName + shaderSuffix + BAKED_POSTFIX + udimSuffix;
-    if (_hashImageNames)
-    {
-        std::stringstream hashStream;
-        hashStream << std::hash<std::string>{}(bakedImageName);
-        bakedImageName = hashStream.str();
-    }
-    return FilePath(bakedImageName + "." + _extension);
+    size_t i = filename.find(var, start);
+    if (var == "$UDIM" && i != string::npos)
+    {
+        size_t udimPrefix = filename.find("$UDIMPREFIX", start);
+        if (i == udimPrefix)
+        {
+            i = filename.find(var, i + 1);
+        }
+    }
+    return i;
+}
+
+FilePath TextureBaker::generateTextureFilename(const StringMap& filenameTemplateMap)
+{
+    string bakedImageName = _textureFilenameTemplate;
+
+    for (auto& pair : filenameTemplateMap)
+    {
+        string replacement = (_texTemplateOverrides.count(pair.first)) ?
+            _texTemplateOverrides[pair.first] : pair.second;
+        replacement = (filenameTemplateMap.at("$UDIM").empty() && pair.first == "$UDIMPREFIX") ?
+            EMPTY_STRING : replacement;
+
+        for (size_t i = 0; (i = findVarInTemplate(bakedImageName, pair.first, i)) != string::npos; i++)
+        {
+            bakedImageName.replace(i, pair.first.length(), replacement);
+        }
+    }
+    return _outputImagePath / bakedImageName;
+}
+
+StringMap TextureBaker::initializeFileTemplateMap(OutputPtr output, NodePtr shader, const string& udim)
+{
+    string inputName = output->getName();
```

Comment:
In this logic, would it be possible to get the unique name of the associated _input_ of the shader node, rather than the modified name of the output?  Since we don't require any particular naming convention for output elements, I can imagine situations where it might have an ambiguous or unclear name, while shader node inputs are required to match the canonical input names in the shader nodedef.

Reply:
I have incorporated the feedback into the code.

Reply:
Thanks for the feedback!